OortCommon.py

```diff
@@ -323,32 +342,93 @@ def configValue(key):
     return gConfig.get(key)
 
 
+def locateConfigDirectory():    
+    # Determine where the oort-config directory lives. The path can be specified in several ways.
+    # Check in order of precedence:
+    # 1. The -c / --config-dir command line argument
+    # 2. The OORTCONFIGDIR environment variable
+    # 3. An .oortconfigpath file in user's home directory, containing the path to oort-config
+    # 4. An .oortconfig directory existing in the user's home directory containing the actual configuration
+    configDirPath = None
+
+    debug("All CLI args: %r" % OortArgs().getArgs())
+
+    # 1. The -c / --config-dir command line argument
+    configDirPath = OortArgs().get('config_dir')
+    debug("configDirPath = %r" % configDirPath)
+    if configDirPath is not None:
+        configDirPath = os.path.expanduser(configDirPath)
+        assert os.path.isdir(configDirPath), configDirPath + " is not a valid oort-config directory path"
+        return configDirPath
+
+    # 2. The OORTCONFIGDIR environment variable
+    configDirPath = os.getenv(CONFIG_DIR_ENV_VAR_NAME)
+    if configDirPath is not None:
+        configDirPath = os.path.expanduser(configDirPath)
+        assert os.path.isdir(configDirPath), "The environment variable " + CONFIG_DIR_ENV_VAR_NAME + " does not contain a valid directory path"
+        return configDirPath
+        
+    # 3. An .oortconfigpath file in user's home directory, containing the path to oort-config
+    configDirPathAlias = os.path.expanduser(os.path.join("~", CONFIG_DIR_ALIAS_FILENAME))
+    if os.path.isfile(configDirPathAlias):
+        with open(configDirPathAlias, "r") as configDirPathAliasFile:
+            configDirPath = configDirPathAliasFile.read().strip()
+    if configDirPath is not None:
+        assert os.path.isdir(configDirPath), "The file ~/" + CONFIG_DIR_ALIAS_FILENAME + " does not contain a valid directory path"
+        return configDirPath
+    
+    # 4. An .oortconfig directory existing in the user's home directory containing the actual configuration
+    configDirPath = os.path.expanduser(os.path.join("~", CONFIG_DIR_DOT_DIRNAME))
+    if os.path.isdir(configDirPath):
+        return configDirPath
+    
+    return configDirPath
```

Comment:
Good catch. The `None` return is intentional, but the calling code would actually fail with a `TypeError` assertion rather than the intended _"you don't have a valid config dir so you can't continue"_ assertion. For reference the call is at lines 459-460:

```
    configDirPath = locateConfigDirectory()
    assert os.path.isdir(configDirPath), "A valid oort-config directory could not be found. You must specify its absolute path via the -c/--config-dir argument, the " + CONFIG_DIR_ENV_VAR_NAME + " environment variable, or the ~/" + CONFIG_DIR_ALIAS_FILENAME + " file, or by storing your configuration in the ~/" + CONFIG_DIR_DOT_DIRNAME + " directory."
```

I'll make the change to catch `None`.